### ISSUE TITLE
Feature Request: Select Different Download Path in Download Dialog

### ISSUE THREAD

**Issue by magi44ken** [TeamNewPipe/NewPipe#2285]
In the Download Dialog, add an option to change the default download path to easily manually save to different folders.

**Comment by pacorrop** [comment]
This is configurable in settings?

I could instead add the possibility of editing the default path by hand (i.e, through the keyboard), since I have Android Nougat and the default path (/storage/emulated/0/Movies/NewPipe) is somehow not reachable by any non-root user. However, /sdcard/Movies/NewPipe is. When I try to set it through the default path browser, /sdcard/ is empty (like if it was not allowed to browse into it). 

I had to manually edit the org.schabi.newpipe.preferences.xml to enable the downloads again.

**Comment by magi44ken** [comment]
Yes you change default path in the settings but I want to be able to do in download dialog. There are times I want to save to different folder beside the default folders in the settings.

**Comment by TobiGr** [comment]
Sounds like a good idea and should be easy to implement

**Comment by sherlockbeard** [comment]
hey can anybody guide on how to make it possible . because it currently get location from environment.
so , my doubt is it possible to select path by sending a intent to file explore . or is there any easy way or other way to this.

**Comment by gkeegan** [comment]
It would probably be best to open the android file manager to select where to download to. Typing in the path may cause issues if NewPipe ever ends up supporting downloading to SD card.

**Comment by sherlockbeard** [comment]
i started the file explore but i am not sure how to get result from it. does anyone have any experience with it.

**Comment by TobiGr** [comment]
Please do not use the Android File picker, but the FilePicker library. You can find its usage here: https://github.com/TeamNewPipe/NewPipe/blob/2d39e65b5cf204ea80d74c7d7919315bbfbd6d8e/app/src/main/java/org/schabi/newpipe/settings/DownloadSettingsFragment.java#L53

However, @kapodamy has just added a storage access framework and other things related to file handling. So it might be clever to build upon the work in #2149 and check which changes @kapodamy made to the way the filepicker is used.

**Comment by sherlockbeard** [comment]
Thanx @TobiGr . I did the fix. using FilePickerActivityHelper

**Comment by MuhammadImran2** [comment]
Hello! I would like to work on this :)

please assign me this issue

**Comment by sherlockbeard** [comment]
damn flashback of 2019 .....

**Comment by MuhammadImran2** [comment]
hahahahah sorry  i do no see time ok

**Comment by sherlockbeard** [comment]
hi @MuhammadImran2  plz free to take this as i can't contribute to it .

**Comment by MuhammadImran2** [comment]
ok @sherlockbeard :heart:

**Comment by TobiGr** [comment]
This needs to wait until after the rewrite or at least until the download handling was refactored

**Comment by MuhammadImran2** [comment]
ok @TobiGr

**Comment by MuhammadImran2** [comment]
ok

**Comment by MuhammadImran2** [comment]
sorry friend  mistakenly click on unAssignees but please assign me thank you

**Comment by MuhammadImran2** [comment]
❤️

**Comment by Gc0rp** [comment]
Can i work on this issue?

**Comment by Gc0rp** [comment]
@ShareASmile could you make me the assignee?
